judge/comments.py

```diff
@@ -72,6 +74,18 @@ def post(self, request, *args, **kwargs):
         if self.is_comment_locked():
             return HttpResponseForbidden()
 
+        if not request.user.is_superuser:
+            user_latest_comments = Comment.get_newest_visible_comments(viewer=request.user,
+                                                                       author=request.user.profile,
```

Comment:
the `get_newest_visible_comments` is costly, don't use that. Can use a simple query set to find the latest comment with the same author is enough

Reply:
I replaced it with a simpler query.

It works a bit differently: instead of getting the latest comment *that the user can see*, this just gets the latest comment. I hope it will do just fine.

Please have a look.